### ISSUE TITLE
Named WINDOW clause not recognized (TSQL)

### ISSUE THREAD

**Issue by cheyney-w** [sqlfluff/sqlfluff#6687]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff reporting unparsable section when encountering a [`WINDOW`](https://learn.microsoft.com/en-us/sql/t-sql/queries/select-window-transact-sql?view=sql-server-ver16) clause using the tsql dialect.

### Expected Behaviour

Proper parsing of the window clause in statements such as this:

```sql
select row_number() over win as x 
from information_schema.tables 
window win as (order by table_name)
```

### Observed Behaviour

```
L:   1 | P:  21 |  PRS | Line 1, Position 21: Found unparsable section: 'over win as x from
                       | information_schema.ta...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

### How to reproduce

```bash
echo 'select row_number() over win as row_num from information_schema.tables window win as (order by table_name)' | sqlfluff parse --dialect=tsql -
```

### Dialect

TSQL

### Version

sqlfluff, version 3.3.1
Python version 3.12.9

### Configuration

None used

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
